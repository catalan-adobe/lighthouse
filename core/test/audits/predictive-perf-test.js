/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import PredictivePerf from '../../audits/predictive-perf.js';
import {getURLArtifactFromDevtoolsLog, readJson} from '../test-utils.js';

const acceptableTrace = readJson('../fixtures/traces/lcp-m78.json', import.meta);
const acceptableDevToolsLog = readJson('../fixtures/traces/lcp-m78.devtools.log.json', import.meta);

describe('Performance: predictive performance audit', () => {
  it('should compute the predicted values', async () => {
    const artifacts = {
      URL: getURLArtifactFromDevtoolsLog(acceptableDevToolsLog),
      GatherContext: {gatherMode: 'navigation'},
      traces: {
        [PredictivePerf.DEFAULT_PASS]: acceptableTrace,
      },
      devtoolsLogs: {
        [PredictivePerf.DEFAULT_PASS]: acceptableDevToolsLog,
      },
    };
    const context = {computedCache: new Map(), settings: {locale: 'en'}};

    const output = await PredictivePerf.audit(artifacts, context);
    expect(output.displayValue).toBeDisplayString('4,460Â ms');
    const metrics = output.details.items[0];
    for (const [key, value] of Object.entries(metrics)) {
      metrics[key] = value === undefined ? value : Math.round(value);
    }
    expect(metrics).toMatchSnapshot();
  });
});
