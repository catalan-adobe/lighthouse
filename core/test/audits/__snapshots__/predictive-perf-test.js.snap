// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Performance: predictive performance audit should compute the predicted values 1`] = `
Object {
  "optimisticFCP": 2294,
  "optimisticFMP": 2294,
  "optimisticLCP": 2294,
  "optimisticSI": 1393,
  "optimisticTTI": 3795,
  "pessimisticFCP": 2294,
  "pessimisticFMP": 3233,
  "pessimisticLCP": 3233,
  "pessimisticSI": 3052,
  "pessimisticTTI": 5119,
  "roughEstimateOfFCP": 2294,
  "roughEstimateOfFMP": 2764,
  "roughEstimateOfLCP": 2764,
  "roughEstimateOfLCPLoadEnd": undefined,
  "roughEstimateOfLCPLoadStart": undefined,
  "roughEstimateOfSI": 3684,
  "roughEstimateOfTTFB": 611,
  "roughEstimateOfTTI": 4457,
}
`;
